### PR TITLE
Perform sync syscall to persist file system contents

### LINKED ISSUE #713: Not able to write to files in the root disk image
I modified helloworld sample so that in addition to reading from "helloworld.txt", it also writes to the file. Note the enclave_config.json file indicates this is a writable disk image:

{
  "args": [
    "app/helloworld"
  ],
  "root": {
    "readonly": false
  }
}
However, after I run the sample, and mount the disk image, file "/app/helloworld.txt" is not updated.

This seems to be blocking code coverage measurement as we can't generate the coverage data files on the disk image.

### LINKED ISSUE #754: File systems unmounted using MNT_DETACH may become inconsistent
We currently unmount the root file system using the `MNT_DETACH` option, otherwise we sometimes observe hanging behaviour (despite the timeout): https://github.com/lsds/sgx-lkl/blob/a1fb63f406557cb02123682efaa48d78018a3421/src/lkl/setup.c#L1276

We should check that this always cleanly unmounts the file systems and does not lead to data loss. It also not clear that the code correctly unmounts the entire file system hierarchy.

### PR DESCRIPTION
**Pull request by prp**
This commit adds a sync syscall to the shutdown sequence, which ensures
that the content of the buffer cache is flushed to disk. It also adds a
test that verifies that a file is persisted across SGX-LKL runs.

Fixes https://github.com/lsds/sgx-lkl/issues/713 and #754.

### CODE REVIEW

File: src/lkl/setup.c [hunk 1]
```diff
@@ -1215,7 +1215,7 @@ static void* lkl_termination_thread(void* args)
     }
 
     // Switch back to root so we can unmount all filesystems
-    SGXLKL_VERBOSE("calling lkl_sys_chdir(/)\n");
+    SGXLKL_VERBOSE("calling lkl_sys_chdir(\"/\")\n");
```
Comment (davidchisnall) [inline src/lkl/setup.c:1218]: Why this change?

**Review by davidchisnall — APPROVED** [review]

File: src/lkl/setup.c (hunk 1 above)
Comment (prp) [reply src/lkl/setup.c:1218]: Consistency with other output.

**Review by jxyang — APPROVED** [review]

**Review by hukoyu — APPROVED** [review]
LGTM
